### DISCUSSION TITLE
How do you host on Tshock??

### DISCUSSION THREAD

**Discussion by ThomasZWasTaken** [Pryaxis/TShock#2144] [General]
Whenever I hear about running a server on tshock they always say the same thing and when I run a server I find out that it only works for local lan. There must be a way to host a server for other to join from other countries. I would really appreciate if you can tell me how.

**Comment by bartico6** [comment]
Depends on if you want to host on your computer or not.

If you want to host from your own computer (not a good idea unless it's a closed server for friends) you have to open and forward ports on your router (according to its manufacturer's notes) as well as make sure server hosting is okay with your internet service provider. That's the cheap/free option, with the exception of the power bill for your computer running, as the server will only run when your computer is on, naturally.

The other option is to rent a server (either just Terraria with a panel, or a dedicated server and set everything up) and for a monthly fee have one running 24/7. This option is obviously not going to be free as you have to pay for the server fees, but its more reliable (and generally the only way to go for public servers, as hosting a public server from home is **a very very very very bad idea**)

**Reply by rautamiekka** [reply]
> [...] have to open and forward ports on your router (according to its manufacturer's notes) as well as make sure server hosting is okay with your internet service provider. [...]

ZeroTier One avoids all that, but the connection might not be as stable (and is fully dependent on ZeroTier) if a direct connection can't be opened.

**Reply by ThomasZWasTaken** [reply]
bartico6 i tried to do the port forward but if someone types my ipv4 address it wont work. its local lan.

**Reply by rautamiekka** [reply]
> bartico6 i tried to do the port forward but if someone types my ipv4 address it wont work. its local lan.

Unless the switch is a managed one and thus is wrongly conffed, your problem is either a misconffed per-computer firewall, or the OS has essentially been bricked (happens to Window$ eventually), to which the only fix is OS reinstall. In Win10's case you might be able to just reset the system using the integrated method instead of doing a reinstall.

**Reply by bartico6** [reply]
> 
> 
> bartico6 i tried to do the port forward but if someone types my ipv4 address it wont work. its local lan.

If you're using Windows, open the port on the firewall. You're also supposed to use the public address (you can get it [here](http://ipv4.whatismyip.akamai.com/)), not local (192.168.x.x) - just in case.

**Reply by bartico6** [reply]
If you're behind a NAT or some sort of weird IP assignment trickery by your ISP you're also out of luck and will need to rent a server as well.

**Reply by rautamiekka** [reply]
> If you're behind a NAT or some sort of weird IP assignment trickery by your ISP you're also out of luck and will need to rent a server as well.

ZeroTier One should still be an option.

**Reply by ThomasZWasTaken** [reply]
what should the nat be set to?? routedwithnat, routedwithoutnat, or bridged

**Reply by rautamiekka** [reply]
> what should the nat be set to?? routedwithnat, routedwithoutnat, or bridged

If NAT causes problems, then disable the NAT but leave routing. Bridged is a bit of a special case.
